public/app/features/query/state/runRequest.ts

```diff
@@ -159,6 +169,7 @@ export function runRequest(
       request.endTime = Date.now();
 
       state = processResponsePacket(packet, state);
+      cache.set(cacheKey, state.panelData);
```

Comment:
I think if we would go with that, we should only cache successful data, i.e. no error responses.